reduction_scripts/reduce_data.py

```diff
@@ -916,43 +916,99 @@ def run_save_3dcube(metadata, prev_suffix, curr_suffix, **args):
-    data_dir = os.path.abspath(sys.argv[1]) + "/"
+    # Validate and process the data_dir
+    data_dir = os.path.abspath(args.data_dir)
+    if not data_dir.endswith("/"):
+        data_dir += "/"
+    print(f"Processing data in directory: {data_dir}")
 
-    naxis2_to_process = 0
-    if len(sys.argv) == 3:
-        naxis2_to_process = int(sys.argv[2])
+    # Handling redcution parameters.
+
+    params_path = {
+        "blue": None,
+        "red": None,
+    }
+
+    # Red
+    if args.red_params:
+        params_path["red"] = os.path.abspath(args.red_params)
+        print(f"Using red parameters from: {params_path['red']}")
+
+    # Blue
+    if args.blue_params:
+        params_path["blue"] = os.path.abspath(args.blue_params)
+        print(f"Using blue parameters from: {params_path['blue']}")
 
     # Classify all raw data (red and blue arm)
+    naxis2_to_process = 0  # TODO is this used in half frame (stellar mode)? Check that and include it as a parameter in the json files if needed.
     obs_metadatas = classify(data_dir, naxis2_to_process)
 
     # Set paths
     reduction_scripts_dir = os.path.dirname(__file__)
     working_dir = os.getcwd()
 
+    # Check grism and observing mode in the first science image of each arm
+    sci_filename_red = obs_metadatas["red"]["sci"][0]["sci"][0] + ".fits"
+    sci_filename_blue = obs_metadatas["blue"]["sci"][0]["sci"][0] + ".fits"
+
+    # Grism: grating are expected to be diferent on each arm. Read them into a dictionary.
+    grism = {
+        "blue": pyfits.getheader(data_dir + sci_filename_blue)["GRATINGB"],
+        "red": pyfits.getheader(data_dir + sci_filename_red)["GRATINGR"],
+    }
+
+    # Observing mode: should be the same in both arm
+    if pywifes.is_nodshuffle(data_dir + sci_filename_red) and pywifes.is_nodshuffle(
+        data_dir + sci_filename_blue
+    ):
+        obs_mode = "ns"
+    elif not pywifes.is_nodshuffle(
+        data_dir + sci_filename_red
+    ) and not pywifes.is_nodshuffle(data_dir + sci_filename_blue):
+        obs_mode = "class"
+    else:
+        print("Inconsistent observation mode beetween arms !")
+
     for arm in obs_metadatas.keys():
         try:
             # ------------------------------------------------------------------------
             #      LOAD JSON FILE WITH USER REDUCTION SETUP
             # ------------------------------------------------------------------------
             obs_metadata = obs_metadatas[arm]
 
-            # Check observing mode
-            sci_filename = obs_metadatas[arm]["sci"][0]["sci"][0] + ".fits"
-            if pywifes.is_nodshuffle(data_dir + sci_filename):
-                obs_mode = "ns"
-            elif pywifes.is_subnodshuffle(data_dir + sci_filename):
-                obs_mode = "ns"
+            # Read the JSON file
+            if params_path[arm] is None:
+                json_path = (
+                    f"./pipeline_params/{arm}/params_{obs_mode}_{grism[arm]}.json"
+                )
```

Comment:
The `README` file should be updated to address these upgrades:
1. Explain that the users can make their custom-made configuration file.
2. Provide the instructions for the new command line args.

Reply:
Done! We should think on putting a JSON file template as reference in a more accessible/transparent path for users. What do you thinks?